tests/apichecks/crds_test.go

```diff
@@ -45,6 +49,107 @@ func TestCRDsDoNotHaveFooUrlRef(t *testing.T) {
 	}
 }
 
+// Enforces acronym capitalization on CRDs
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions
+// All letters in the acronym should have the same case, using the appropriate case for the situation.
+// For example, at the beginning of a field name, the acronym should be all lowercase, such as "httpGet".
+// Where used as a constant, all letters should be uppercase, such as "TCP" or "UDP".
+func TestCRDsAcronyms(t *testing.T) {
+	crds, err := crdloader.LoadAllCRDs()
+	if err != nil {
+		t.Fatalf("error loading crds: %v", err)
+	}
+
+	var errs []string
+	for _, crd := range crds {
+		for _, version := range crd.Spec.Versions {
+			visitCRDVersion(version, func(fieldPath string) {
+
+				tokens := splitCamelCase(fieldPath)
+
+				// Special cases for common acronyms
+				for i, token := range tokens {
+					isAcronym := false
+
+					switch strings.ToLower(token) {
+					case "http", "https", "ssh", "tls", "udp", "tcp":
+						isAcronym = true
+					case "api":
+						isAcronym = true
+
+					case "ipv4", "ipv6", "ip", "cidr", "bgp":
+						isAcronym = true
+
+					case "id":
+						isAcronym = true
+
+					case "url":
+						isAcronym = true
+					case "cdn":
+						isAcronym = true
+					case "nat":
+						isAcronym = true
+					case "x509":
+						isAcronym = true
+					case "sso":
+						isAcronym = true
+					case "oauth2", "oidc":
+						isAcronym = true
+					case "iap":
+						isAcronym = true
+					case "os":
+						isAcronym = true
+					}
+
+					// TODO: Ips, Cidrs
+
+					// TODO: Src / Dest
+
+					if isAcronym {
+						if i == 0 {
+							tokens[i] = strings.ToLower(token)
+						} else {
+							tokens[i] = strings.ToUpper(token)
+						}
+					}
+				}
+
+				corrected := strings.Join(tokens, "")
+
+				if corrected != fieldPath {
+					errs = append(errs, fmt.Sprintf("[acronyms]: field %q in %q (%v) should be %q", fieldPath, crd.Name, version.Name, corrected))
+				}
+			})
+		}
+	}
+
+	sort.Strings(errs)
+
+	want := strings.Join(errs, "\n")
+
+	test.CompareGoldenFile(t, "errata/acronyms.txt", want)
```

Comment:
Cool example! I would argue to rename it to testdata/...  (what does errata mean?)

Reply:
So it used to be that fixing mistakes in a printed book was expensive, so they would instead include a sheet with significant errors listed.   I found this fun picture: https://analoggamestudies.org/wordpress/wp-content/uploads/2014/11/lit_errata.jpg

Now it's mostly used to highlight changes in things like chip specifications, it's sort of like a changelog except the changes are to the description rather than to the chip itself - it's infeasible to change the chip.  e.g. https://www.infineon.com/dgdl/Infineon-XMC4100_XMC4200_BA_Errata_Sheet-ES-v01_03-EN.pdf?fileId=5546d46259b0420a0159d50916760b89

How about testdata/exceptions ?

Reply:
(The argument _against_ testdata BTW is that although we're implementing this in a test, it could also be a linter CLI.  But I agree that errata is pretty obscure at this point, given that "document tabs" are now also considered obscure!)

Reply:
I LOVE this name. in case you haven't changed. Let me get them in!

Reply:
oh no... do you want to change it back😄 ? 

I'll leave it to you to decide.